lib/cassava/client.rb

```diff
@@ -0,0 +1,166 @@
+module Cassava
+  class Client
+    attr_reader :session
+
+    # @param [Cassandra::Session] The session object
+    def initialize(session)
+      @session = session
+    end
+
+    def insert_async(table, data)
+      session.execute_async(insert_statement(table, data), :arguments => data.values)
+    end
+
+    def insert(table, data)
+      session.execute(insert_statement(table, data), :arguments => data.values)
+    end
+
+    # @param [Symbol] table the table name
+    # @param [Array<Symbol>] An optional list of column names (as symbols), to only select those columns
+    # @return [StatementBuilder] A select builder representing the partially completed statement.
+    def select(table, columns = nil)
```

Comment:
Im not sure that returning a StatementBuilder is the correct layer of abstraction here. It might be nice to have this actually execute the query. This could lead to a construct_select which returns the statement_builder IE for generating multiple selects (use cases: application-side search, multi-part reads, PBDS, etc)